File: src/modifiers/navigationBarBackButtonHidden.ts
import { createModifier } from './createModifier';

/// SwiftUI `.navigationBarBackButtonHidden(_:)`. Hides the automatic back
/// button on a view pushed inside a `<NavigationStack>`. Defaults to
/// hiding it (`value: true`); pass `false` to restore the default button.
export function navigationBarBackButtonHidden(value: boolean = true) {
  return createModifier('navigationBarBackButtonHidden', { value });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/NavigationBarBackButtonHidden.swift
import SwiftUI

/// SwiftUI `.navigationBarBackButtonHidden(_:)`. Hides the automatic back
/// button on a view pushed inside a `NavigationStack`. Defaults to hiding
/// (`value` absent → true) to match the JS factory's default argument.
enum RNWNavigationBarBackButtonHiddenModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("navigationBarBackButtonHidden") { view, params, _ in
            let hidden = params.bool("value") ?? true
            return AnyView(view.navigationBarBackButtonHidden(hidden))
        }
    }
}
